src/Bundle/DependencyInjection/SyliusMailerExtension.php

```diff
@@ -44,7 +44,8 @@ protected function loadInternal(array $mergedConfig, ContainerBuilder $container
 
     private function configureSenderAdapter(array $mergedConfig, ContainerBuilder $container): void
     {
-        if (!ContainerBuilder::willBeAvailable('swiftmailer/swiftmailer', \Swift::class, ['symfony/swiftmailer-bundle'])) {
+        $bundles = $container->hasParameter('kernel.bundles') ? $container->getParameter('kernel.bundles') : [];
+        if (!is_array($bundles) || !isset($bundles['SwiftmailerBundle'])) {
```

Comment:
I wonder, is there any way we could have a unified way to check it for both Swiftmailer and Symfony Mailer? On the other words: why does `willBeAvailable` work for SymfonyMailer but not for Swiftmailer? 🤔

Reply:
It doesn't 100% work for either dependency at this point in the compile process.

Basing the Swift check on whether the bundle is installed will get a bit closer to being truthy than whether the library is available in the production environment.  Through the dependency chains, it'd be possible to have the `swiftmailer/swiftmailer` library and `symfony/swiftmailer-bundle` package installed, but if the bundle isn't activated in the kernel, then `ContainerBuilder::willBeAvailable()` would be true and the `kernel.bundles` parameter check would be false.

For the most reliable solution, you need to use a compiler pass that can accurately check if the dependent services of all of the adapters are available (Twig renderer, Swift sender, Symfony sender).  Twig and Swift both have the same kind of convention in that they rely on a separate library and bundle being installed and enabled, the Symfony mailer can be disabled through config on the FrameworkBundle.